enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,501 @@
+annotations). This example would also instruct ExternalDNS to create DNS Records for CR instances of type `dnsrecords` (not a part of OCP 4.9 implementation).
+Finally, this example would instruct ExternalDNS to create DNS Records in the given Private zone for Service resource instances in namespace `foo`.
+
+ExternalDNS's `--annotation-filter` option filters source resource instances via annotations using label selector semantics, which is somewhat unexpected.
+Note that the upstream CRD source implementation actually filters based on resource labels. If upstream ExternalDNS is open to the idea of implementing
+proper label selector logic for all source types (instead of label selector based annotations), work may need to be done in the upstream ExternalDNS
+community accordingly.
+
+As a side note, ExternalDNS deployments will be configured to run ExternalDNS pods on cluster control-plane nodes, as is common practice
+for add-on control-plane components.
+
+### ExternalDNS CRD Source Support (Post OCP 4.9)
+
+As an example, here is what a compatible DNSRecord Custom Resource instance may look like:
+
+```yaml
+apiVersion: operator.externaldns.k8s.io/v1alpha1
+kind: DNSRecord
+metadata:
+  name: default-wildcard
+  namespace: openshift-ingress-operator
+spec:
+  dnsName: 'foo.example.com'
+  providerSpecific:
+  - name: my-provider-setting-name
+    value: my-provider-setting-value
+  recordTTL: 30
+  recordType: A
+  targets:
+  - 1.2.3.4
+status:
+  observedGeneration: 1
+```
+
+This manifest creates a DNS A record for the hostname `foo.example.com` that resolves to address `1.2.3.4` in the configured
+provider, based on the resource's `dnsName` field value. Additional `spec` and/or `status` fields may be introduced based on requirements.
+Note the optional `providerSpecific` field, in which name/value pairs can be added as necessary.
+
+Note that currently, the ExternalDNS CRD source implementation only updates the `observedGeneration` status subfield.
+An [issue has been opened upstream](https://github.com/kubernetes-sigs/external-dns/issues/2092) to better understand
+upstream's intentions with CRD source status fields. ExternalDNS does provide adequate debug logs, should a record fail to update,
+in addition to a few [high-level metrics](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go).
+However, status fields to reflect the availability of the requested record in the given provider would be useful for other controllers running
+in a cluster, as well as for cluster administrators who want to verify that their requested resources have been created properly.
+
+Currently, the upstream ExternalDNS CRD source only supports record creation for A or CNAME type records. See
+[upstream issue 164](https://github.com/kubernetes-sigs/external-dns/issues/164) for more context to that end. Contributions may need to be made to
```

Comment:
Upstream issue 164 seems to be related to leader election. I suspect you are referencing the wrong issue here.

Reply:
Looks like I accidentally dropped the 7 from https://github.com/kubernetes-sigs/external-dns/issues/1647.

Reply:
Fixed. Thanks!